Installing-Necessary-CLI-Tools.md

```diff
@@ -0,0 +1,35 @@
+# Installing CLI Tools Developers Should Not Live Without (Linux and Mac)
+
+This article is a short description on how to install key command line utilities that developers use everyday on Macintosh and Linux environments. The main CLI tools that will be shown are: Homebrew (Mac), Node and NPM, Bower, Gulp and Grunt.
+
+### Installing Homebrew (Macintosh Systems)
+
+Homebrew is ‘The missing package manager for OS X’. It is a great tool for downloading and installing packages straight from your command line. This is not needed on Linux distributions because they already have package managers installed by default such as `apt-get` or `pacman`. To install Homebrew, simply paste the following command in your terminal:
+* `/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`
+
+### Installing NPM
+
+`NPM`, or Node Package Manager, is another useful package manager for downloading mostly web tools. Downloading `NPM` will also install the Node.js framework.
+
+##### Mac:
+* Using `Homebrew` type: `brew install node` and `NPM` should have been installed.
+
+##### Linux:
```

Comment:
Node JS should not be installed via package manager. A lot of programs that depend on Node JS, requires Node JS to be invoked as `$ node` in terminal. So, add some guide to update symlinks properly.

The recommended way to install Node JS is either compile directly from source; or install `nvm` first, then go with `nvm use <version_number>`. Given how quickly Node JS has been releasing new versions, it's better to go with second one to stay on top of things, even when you need LTS versions.